packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -117,14 +117,22 @@ export class MediaPlaceholder extends Component {
 			multiple,
 			onError,
 			onSelect,
-			value = [],
 		} = this.props;
 		let setMedia;
 		if ( multiple ) {
 			if ( addToGallery ) {
-				const currentValue = value;
+				let lastMediaPassed = [];
 				setMedia = ( newMedia ) => {
-					onSelect( currentValue.concat( newMedia ) );
+					const currentMedia = this.props.value || [];
+					// Remove lastMediaPassed from currentMedia.
+					const mediaToReturn = currentMedia.filter( ( item ) => {
+						return ! lastMediaPassed.some( ( temporaryMediaURL ) => temporaryMediaURL === item.url );
+					} );
+					// Reset lastMediaPassed and set it with urls from newMedia.
+					lastMediaPassed = [];
+					newMedia.forEach( ( media ) => lastMediaPassed.push( media.url ) );
+
+					onSelect( mediaToReturn.concat( newMedia ) );
```

Comment:
Could we add a comment here to explain what the variable means -- like why we use `mediaToReturn` instead of `lastMediaPassed` or something? It just took me a while to understand what the different variables were doing again, so a comment might help make it clearer to new folks

Reply:
I added a few more comments and renamed `mediaToReturn` as well.  Hopefully this reads a bit easier now?

Reply:
Yeah, that definitely reads better, thanks! :D